distributed/client.py

```diff
@@ -2551,12 +2556,24 @@ def add(self, future):
     def __iter__(self):
         return self
 
+    def __aiter__(self):
+        return self
+
     def __next__(self):
         with self.lock:
             if not self.futures and self.queue.empty():
                 raise StopIteration()
         return self.queue.get()
 
+    @gen.coroutine
+    def __anext__(self):
+        if not self.futures and self.queue.empty():
+            raise StopAsyncIteration  # flake8: noqa
+        while self.queue.empty():
+            yield self.event.wait()
```

Comment:
This event is set once and never cleared, so once you've read one thing from the queue this turns into a busy loop. Maybe that's starving out some other thread that needs to run on windows? 

It looks like you want a `tornado.locks.Condition` instead.